src/licensedcode/detection.py

```diff
@@ -263,6 +370,278 @@ def matched_text(
+            if is_license_ref:
+                spdx_url = SCANCODE_LICENSE_TEXT_URL.format(lic.key)
+            else:
+                spdx_key = lic.spdx_license_key.rstrip('+')
+                spdx_url = SPDX_LICENSE_URL.format(spdx_key)
+        else:
+            spdx_url = ''
+        result['spdx_url'] = spdx_url
+        result['start_line'] = match.start_line
+        result['end_line'] = match.end_line
+        matched_rule = result['matched_rule'] = {}
+        matched_rule['identifier'] = match.rule.identifier
+        matched_rule['license_expression'] = match.rule.license_expression
+        matched_rule['licenses'] = match.rule.license_keys()
+        matched_rule['referenced_filenames'] = match.rule.referenced_filenames
+        matched_rule['is_license_text'] = match.rule.is_license_text
+        matched_rule['is_license_notice'] = match.rule.is_license_notice
+        matched_rule['is_license_reference'] = match.rule.is_license_reference
+        matched_rule['is_license_tag'] = match.rule.is_license_tag
+        matched_rule['is_license_intro'] = match.rule.is_license_intro
+        matched_rule['has_unknown'] = match.rule.has_unknown
+        matched_rule['matcher'] = match.matcher
+        matched_rule['rule_length'] = match.rule.length
+        matched_rule['matched_length'] = match.len()
+        matched_rule['match_coverage'] = match.coverage()
+        matched_rule['rule_relevance'] = match.rule.relevance
+        # FIXME: for sanity this should always be included?????
+        if include_text:
+            result['matched_text'] = matched_text
+    return detected_licenses
+
+
+def matches_compact_repr(matches):
+    """
+    Return a string representing a list of license matches in a compact way.
+    """
+    return ', '.join(m.rule.identifier for m in matches)
+
+
+def is_correct_detection(license_matches):
+    """
+    Return True if all the license matches in a file-region are correct
+    license detections, as they are either SPDX license tags, or the file content has
+    a exact match with a license hash.
+
+    :param license_matches: list
+        List of LicenseMatch.
+    """
+    matchers = (license_match.matcher for license_match in license_matches)
+    return (
+        all(matcher in ("1-hash", "1-spdx-id") for matcher in matchers)
+        and not has_unknown_matches(license_matches)
+    )
+
+
+def is_match_coverage_less_than_threshold(license_matches, threshold):
+    """
+    Returns True if any of the license matches in a file-region has a `match_coverage`
+    value below the threshold.
+
+    :param license_matches: list
+        List of LicenseMatch.
+    :param threshold: int
+        A `match_coverage` threshold value in between 0-100
+    """
+    coverage_values = (
+        license_match.coverage() for license_match in license_matches
+    )
+    return any(coverage_value < threshold for coverage_value in coverage_values)
```

Comment:
IMHO you can conflate the two lines without loosing redability
```suggestion
    return any(license_match.coverage() < threshold for for license_match in license_matches)
```

Reply:
Updated. Thanks!